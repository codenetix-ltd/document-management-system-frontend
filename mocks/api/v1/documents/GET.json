{
  "links": {
    "first": "/api/v1/documents?page=1",
    "last": "/api/v1/documents?page=4",
    "prev": null,
    "next": "/api/v1/documents?page=2"
  },
  "meta": {
    "currentPage": 1,
    "from": 1,
    "lastPage": 8,
    "path": "/api/v1/documents",
    "perPage": 15,
    "to": 15,
    "total": 51
  },
  "data": [
    {
      "name": "Wooden sticks Magnum",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 1,
      "templateId": 1,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Classic",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 2,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight 114 mm",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 3,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks 114x10x2 mm",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 4,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly 93 mm RPK",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 5,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly 93 mm RPK",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 6,
      "templateId": 6,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Mag",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 7,
      "templateId": 1,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Class",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 8,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 9,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 10,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 11,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 12,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 13,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 14,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 15,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    }
  ]
}
