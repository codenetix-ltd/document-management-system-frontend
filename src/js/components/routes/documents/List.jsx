import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import moment from 'moment';
import first from 'lodash/first';
import pick from 'lodash/pick';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';

import Prompt from 'Components/common/Prompt';
import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';
import SelectableTable from 'Components/common/SelectableTable';

import { DataLink } from 'Components/common/dataControls';

import {
  $$documentArchive,
  $$documentDelete,
  $$documentsFetch
} from 'Store/thunks/documents';

import { $setSelectedDocuments } from 'Store/actions';

import FiltersWrapper from './partials/FiltersWrapper';
import DocumentActions from './partials/DocumentActions';
import ArchiveModalContent from './partials/ArchiveModalContent';

@autobind
export class DocumentsList extends Component {
  static defaultProps = {
    substituteDocument: null
  };

  static propTypes = {
    substituteDocument: PropTypes.any,
    documents: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  componentWillUnmount() {
    this.clearSelection();
  }

  onFetchData(tableProps) {
    let orderBy = 'id';
    let sortedBy = 'desc';
    const { dispatch } = this.props;
    const { page, sorted } = tableProps;
    const sortData = first(sorted);
    if (sortData) {
      orderBy = sortData['id'];
      sortedBy = sortData['desc'] ? 'desc' : 'asc';
    }
    $$documentsFetch(dispatch, {
      page: page + 1,
      orderBy,
      sortedBy
    }, this.clearSelection);
  }

  onSelect(selected) {
    this.props.dispatch($setSelectedDocuments(selected));
  }

  onArchive({ value: { actualVersion } }) {
    const { dispatch } = this.props;
    this.prompt.show({
      width: 555,
      body: <ArchiveModalContent />,
      confirmText: 'Archive',
      onConfirm: close => {
        const { substituteDocument: { id } } = this.props;
        $$documentArchive(dispatch, actualVersion, id, close);
      }
    });
  }

  onDelete({ value: { actualVersion } }) {
    const { dispatch } = this.props;
    this.prompt.show({
      body: `Do you really want to delete document ${actualVersion.name}?`,
      confirmText: 'Delete',
      onConfirm: close => $$documentDelete(dispatch, actualVersion, close)
    });
  }

  clearSelection() {
    this.props.dispatch($setSelectedDocuments([]));
  }

  breadcrumbs = [
    { pageName: 'Documents list', pageLink: '/documents/list', iconCls: 'fa fa-list' }
  ];

  columns = [
    {
      Header: 'Id',
      accessor: 'id',
      maxWidth: 65,
    }, {
      Header: 'Name',
      accessor: 'actualVersion.name',
    }, {
      Header: 'Owner',
      accessor: 'owner.fullName',
    }, {
      Header: 'Template ',
      accessor: 'actualVersion.template.name',
    }, {
      Header: 'Created at',
      accessor: 'createdAt',
      Cell: ({ value }) => moment.unix(value).format('YYYY/MM/DD')
    }, {
      Header: 'Updated at',
      accessor: 'updatedAt',
      Cell: ({ value }) => moment.unix(value).format('YYYY/MM/DD')
    }, {
      Header: 'Actions',
      accessor: '',
      maxWidth: 122,
      sortable: false,
      Cell: rowData => {
        const viewLink = `/documents/view/${rowData.value.id}`;
        const editLink = `/documents/${rowData.value.id}`;
        return (
          <div>
            <Link to={viewLink} className="btn btn-primary btn-xs">
              <i className="fa fa-eye" />
            </Link>
            &nbsp;
            <Link to={editLink} className="btn-success btn btn-xs">
              <i className="fa fa-edit" />
            </Link>
            &nbsp;
            <DataLink cls="btn btn btn-warning btn-xs" data={rowData} onClick={this.onArchive}>
              <i className="fa fa-book" />
            </DataLink>
            &nbsp;
            <DataLink cls="btn-danger btn btn-xs" data={rowData} onClick={this.onDelete}>
              <i className="fa fa-trash" />
            </DataLink>
          </div>
        );
      }
    }
  ];

  render() {
    const { documents, loading } = this.props;
    return (
      <div>
        <ContentHeader title="Documents" breadcrumbs={this.breadcrumbs} />
        <ContentWrapper boxClass="box-success">
          <div className="box-body">
            <FiltersWrapper />
            <hr />
            <DocumentActions prompt={this.prompt} />
            <div className="box-body">
              <div className="row">
                <div className="col-sm-12">
                  <AlertMessage />
                  <ErrorMessage />
                  <SelectableTable
                    manual
                    className="-striped"
                    columns={[...this.columns]}
                    data={documents.list}
                    pages={documents.lastPage}
                    loading={loading}
                    onFetchData={this.onFetchData}
                    onSelect={this.onSelect}
                    defaultPageSize={15}
                    PaginationComponent={Pagination}
                    multiSort={false}
                    resizable={false}
                  />
                </div>
              </div>
            </div>
          </div>
        </ContentWrapper>
        <Prompt ref={p => { this.prompt = p; }} />
      </div>
    );
  }
}

const mapStateToProps = (state) => pick(state, ['documents', 'loading', 'substituteDocument']);

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(DocumentsList);
