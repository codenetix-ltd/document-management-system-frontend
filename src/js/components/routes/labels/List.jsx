import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import ReactTable from 'react-table';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';
import ActionsCell from 'Components/partials/ActionsCell';

import Prompt from 'Components/common/Prompt';
import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';

import {
  $$labelsFetch,
  $$labelDelete
} from 'Store/thunks/labels';

@autobind
export class LabelsList extends Component {
  static propTypes = {
    labels: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  onFetchData({ page }) {
    $$labelsFetch(this.props.dispatch, page + 1);
  }

  onDelete({ value }) {
    const { fullName } = value;
    const { dispatch } = this.props;
    this.prompt.show({
      body: `Do you really want to delete label ${fullName}?`,
      onConfirm: close => $$labelDelete(dispatch, value, close)
    });
  }

  render() {
    const { labels, loading } = this.props;

    const columns = [
      {
        Header: 'Id',
        accessor: 'id',
        width: 116,
      }, {
        Header: 'Name',
        accessor: 'name',
      }, {
        Header: 'Actions',
        accessor: '',
        width: 262,
        sortable: false,
        Cell: (rowData) => {
          const editLink = `/labels/${rowData.value.id}`;
          return <ActionsCell editLink={editLink} rowData={rowData} onDelete={this.onDelete} />;
        }
      }
    ];

    const breadcrumbs = [
      { pageName: 'Labels', pageLink: '/labels/list', iconCls: 'fa fa-tags' }
    ];

    return (
      <div>
        <ContentHeader title="Labels" breadcrumbs={breadcrumbs} />
        <ContentWrapper boxClass="">
          <div className="box-header clearfix">
            <Link to="/labels" className="btn btn-success btn-xs pull-right">
              <i className="fa fa-plus" /> Add label
            </Link>
          </div>
          <div className="box-body">
            <div className="row">
              <div className="col-sm-12">
                <AlertMessage />
                <ErrorMessage />
                <ReactTable
                  manual
                  className="-striped"
                  columns={columns}
                  data={labels.list}
                  pages={labels.lastPage}
                  loading={loading}
                  defaultPageSize={15}
                  onFetchData={this.onFetchData}
                  PaginationComponent={Pagination}
                />
              </div>
            </div>
          </div>
        </ContentWrapper>
        <Prompt ref={p => { this.prompt = p; }} confirmText="Delete" />
      </div>
    );
  }
}

const mapStateToProps = ({ labels, loading }) => ({ labels, loading });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(LabelsList);
