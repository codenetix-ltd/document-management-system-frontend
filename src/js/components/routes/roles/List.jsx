import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import ReactTable from 'react-table';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';
import ActionsCell from 'Components/partials/ActionsCell';

import Prompt from 'Components/common/Prompt';
import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';

import {
  $$rolesFetch,
  $$roleDelete
} from 'Store/thunks/roles';

@autobind
export class RolesList extends Component {
  static propTypes = {
    roles: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  onFetchData({ page }) {
    $$rolesFetch(this.props.dispatch, page + 1);
  }

  onDelete({ value }) {
    const { fullName } = value;
    const { dispatch } = this.props;
    this.prompt.show({
      body: `Do you really want to delete role ${fullName}?`,
      onConfirm: close => $$roleDelete(dispatch, value, close)
    });
  }

  columns = [
    {
      Header: 'Id',
      accessor: 'id',
      width: 116,
    }, {
      Header: 'Name',
      accessor: 'name',
    }, {
      Header: 'Actions',
      accessor: '',
      width: 262,
      sortable: false,
      Cell: (rowData) => {
        const editLink = `/roles/${rowData.value.id}`;
        return <ActionsCell editLink={editLink} rowData={rowData} onDelete={this.onDelete} />;
      }
    }
  ];

  breadcrumbs = [
    { pageName: 'Roles', pageLink: '/roles/list', iconCls: 'fa fa-tags' }
  ];

  render() {
    const { roles, loading } = this.props;
    return (
      <div>
        <ContentHeader title="Roles" breadcrumbs={this.breadcrumbs} />
        <ContentWrapper boxClass="">
          <div className="box-header clearfix">
            <Link to="/roles" className="btn btn-success btn-xs pull-right">
              <i className="fa fa-plus" /> Add role
            </Link>
          </div>
          <div className="box-body">
            <div className="row">
              <div className="col-sm-12">
                <AlertMessage />
                <ErrorMessage />
                <ReactTable
                  manual
                  className="-striped"
                  columns={[...this.columns]}
                  data={roles.list}
                  pages={roles.lastPage}
                  loading={loading}
                  defaultPageSize={15}
                  onFetchData={this.onFetchData}
                  PaginationComponent={Pagination}
                />
              </div>
            </div>
          </div>
        </ContentWrapper>
        <Prompt ref={p => { this.prompt = p; }} confirmText="Delete" />
      </div>
    );
  }
}

const mapStateToProps = ({ roles, loading }) => ({ roles, loading });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(RolesList);
