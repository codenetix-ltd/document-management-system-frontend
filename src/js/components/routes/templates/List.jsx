import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import ReactTable from 'react-table';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';
import ActionsCell from 'Components/partials/ActionsCell';

import Prompt from 'Components/common/Prompt';
import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';

import {
  $$templatesFetch,
  $$templateDelete
} from 'Store/thunks/templates';

@autobind
export class TemplatesList extends Component {
  static propTypes = {
    templates: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  onFetchData({ page }) {
    $$templatesFetch(this.props.dispatch, page + 1);
  }

  onDelete({ value }) {
    const { name } = value;
    const { dispatch } = this.props;
    this.prompt.show({
      body: `Do you really want to delete template ${name}?`,
      onConfirm: close => $$templateDelete(dispatch, value, close)
    });
  }

  render() {
    const columns = [
      {
        Header: 'Id',
        accessor: 'id',
        width: 250,
      }, {
        Header: 'Name',
        accessor: 'name',
      }, {
        Header: 'Actions',
        accessor: '',
        width: 262,
        sortable: false,
        Cell: (rowData) => {
          const editLink = `/templates/${rowData.value.id}`;
          return <ActionsCell editLink={editLink} rowData={rowData} onDelete={this.onDelete} />;
        },
      },
    ];

    const breadcrumbs = [
      { pageName: 'Templates', pageLink: '/templates/list', iconCls: 'fa fa-copy' }
    ];

    return (
      <div>
        <ContentHeader title="Templates" breadcrumbs={breadcrumbs} />
        <ContentWrapper boxClass="">
          <div className="box-header clearfix">
            <Link to="/users" className="btn btn-success btn-xs pull-right">
              <i className="fa fa-plus" /> Add template
            </Link>
          </div>
          <div className="box-body">
            <div className="row">
              <div className="col-sm-12">
                <AlertMessage />
                <ErrorMessage />
                <ReactTable
                  manual
                  className="-striped"
                  columns={columns}
                  data={this.props.templates.list}
                  pages={this.props.templates.lastPage}
                  loading={this.props.loading}
                  defaultPageSize={15}
                  onFetchData={this.onFetchData}
                  PaginationComponent={Pagination}
                />
              </div>
            </div>
          </div>
        </ContentWrapper>
        <Prompt ref={p => { this.prompt = p; }} confirmText="Delete" />
      </div>
    );
  }
}

const mapStateToProps = ({ templates, loading }) => ({ templates, loading });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(TemplatesList);
