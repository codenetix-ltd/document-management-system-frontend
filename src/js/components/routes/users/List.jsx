import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import ReactTable from 'react-table';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';
import ActionsCell from 'Components/partials/ActionsCell';

import Prompt from 'Components/common/Prompt';
import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';

import {
  $$usersFetch,
  $$userDelete
} from 'Store/thunks/users';

@autobind
export class UsersList extends Component {
  static propTypes = {
    users: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  onFetchData({ page }) {
    $$usersFetch(this.props.dispatch, page + 1);
  }

  onDelete({ value }) {
    const { fullName } = value;
    const { dispatch } = this.props;
    this.prompt.show({
      body: `Do you really want to delete user ${fullName}?`,
      onConfirm: close => $$userDelete(dispatch, value, close)
    });
  }

  columns = [
    {
      Header: 'Id',
      accessor: 'id',
      width: 116,
    }, {
      Header: 'Full Name',
      accessor: 'fullName',
    }, {
      Header: 'Email',
      accessor: 'email',
    }, {
      Header: 'Actions',
      accessor: '',
      width: 262,
      sortable: false,
      Cell: (rowData) => {
        const editLink = `/users/${rowData.value.id}`;
        return <ActionsCell editLink={editLink} rowData={rowData} onDelete={this.onDelete} />;
      }
    }
  ];

  breadcrumbs = [
    { pageName: 'Users', pageLink: '/users/list', iconCls: 'fa fa-users' }
  ];

  render() {
    const { users, loading } = this.props;
    return (
      <div>
        <ContentHeader title="Users" breadcrumbs={this.breadcrumbs} />
        <ContentWrapper boxClass="">
          <div className="box-header clearfix">
            <Link to="/users" className="btn btn-success btn-xs pull-right">
              <i className="fa fa-plus" /> Add user
            </Link>
          </div>
          <div className="box-body">
            <div className="row">
              <div className="col-sm-12">
                <AlertMessage />
                <ErrorMessage />
                <ReactTable
                  manual
                  className="-striped"
                  columns={[...this.columns]}
                  data={users.list}
                  pages={users.lastPage}
                  loading={loading}
                  defaultPageSize={15}
                  onFetchData={this.onFetchData}
                  PaginationComponent={Pagination}
                />
              </div>
            </div>
          </div>
        </ContentWrapper>
        <Prompt ref={p => { this.prompt = p; }} confirmText="Delete" />
      </div>
    );
  }
}

const mapStateToProps = ({ users, loading }) => ({ users, loading });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(UsersList);
