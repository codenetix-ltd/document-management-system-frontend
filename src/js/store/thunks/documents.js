import { API } from 'Config';
import pick from 'lodash/pick';
import pickBy from 'lodash/pickBy';
import identity from 'lodash/identity';

import axios from 'Services/request';

import {
  $loading,
  $document,
  $updateDocument,
  $documentsList,
  $setComparedDocuments
} from 'Store/actions';

import { $$messageSet } from 'Store/thunks/message';
import { $$errorSet } from 'Store/thunks/error';

import initialState from 'Store/reducers/initialState.json';

const fn = () => {};

export const $$documentsFetch = (dispatch, params, callback = fn) => {
  const picked = pick(params, ['page', 'sortField', 'sortDirection']);
  const filters = pickBy(params.filterSet, identity);
  dispatch($loading(true));
  axios.get(API.documents, {
    params: { ...picked, ...filters }
  }).then(({ data }) => {
    dispatch($documentsList({
      list: data.data,
      lastPage: data.meta.lastPage,
      ...picked
    }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$documentFetch = (dispatch, documentID, callback = fn) => {
  dispatch($loading(true));
  axios.get(`${API.documents}/${documentID}`).then(({ data }) => {
    dispatch($document({ ...data }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$documentUpdate = (dispatch, part) => {
  dispatch($updateDocument(part));
};

export const $$documentReset = (dispatch) => {
  dispatch($document(initialState.document));
};

export const $$documentDelete = (dispatch, { id, name }, close = fn) => {
  axios.delete(`${API.documents}/${id}`).then(() => {
    $$documentsFetch(dispatch, 1);
    $$messageSet(dispatch, {
      type: 'success',
      text: `The document ${name} was successfully deleted.`
    });
    close();
  }).catch(err => {
    console.trace(err);
    close();
    $$errorSet(dispatch, err);
  });
};

export const $$documentsMassDelete = (dispatch, ids, close = fn) => {
  const p = Promise.all(ids.map(id => axios.delete(`${API.documents}/${id}`)));
  p.then(() => {
    $$documentsFetch(dispatch, 1);
    $$messageSet(dispatch, {
      type: 'success',
      text: 'Selected documents were successfully deleted.'
    });
    close();
  }).catch(err => {
    console.trace(err);
    close();
    $$errorSet(dispatch, err);
  });
};

export const $$comparedDocumentsFetch = (dispatch, ids, callback = fn) => {
  const p = Promise.all(ids.map(id => axios.get(`${API.documents}/${id}`)));
  p.then(d => {
    dispatch($setComparedDocuments(d.map(({ data }) => data)));
    callback(d);
  }).catch(err => {
    console.trace(err);
  });
};
