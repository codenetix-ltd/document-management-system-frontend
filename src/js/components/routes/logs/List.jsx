import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';
import moment from 'moment';
import first from 'lodash/first';

import ReactTable from 'react-table';

import ContentHeader from 'Components/ContentHeader';
import ContentWrapper from 'Components/ContentWrapper';
import Pagination from 'Components/partials/Pagination';

import AlertMessage from 'Components/common/AlertMessage';
import ErrorMessage from 'Components/common/ErrorMessage';

import { $$logsFetch } from 'Store/thunks/logs';

import FiltersWrapper from './partials/FiltersWrapper';

@autobind
export class LogsList extends Component {
  static propTypes = {
    logs: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  onFetchData(tableProps) {
    let orderBy = 'id';
    let sortedBy = 'desc';
    const { dispatch } = this.props;
    const { page, sorted } = tableProps;
    const sortData = first(sorted);
    if (sortData) {
      orderBy = sortData['id'];
      sortedBy = sortData['desc'] ? 'desc' : 'asc';
    }
    $$logsFetch(dispatch, {
      page: page + 1,
      orderBy,
      sortedBy
    });
  }

  columns = [
    {
      Header: 'Id',
      accessor: 'id',
      maxWidth: 100
    }, {
      Header: 'User',
      accessor: 'user.fullName'
    }, {
      Header: 'Action',
      accessor: 'action'
    }, {
      Header: 'Link',
      accessor: 'link',
      Cell: ({ value }) => {
        if (value) {
          return (
            <Link to={value.url}>{value.title}</Link>
          );
        }
        return <span />;
      }
    }, {
      Header: 'Type',
      accessor: 'type'
    }, {
      Header: 'Created at',
      accessor: 'createdAt',
      Cell: ({ value }) => moment.unix(value).format('YYYY/MM/DD')
    }
  ];

  breadcrumbs = [
    { pageName: 'Logs list', pageLink: '/logs/list', iconCls: 'fa fa-list' }
  ];

  render() {
    const { logs, loading } = this.props;
    return (
      <div>
        <ContentHeader title="Logs" breadcrumbs={this.breadcrumbs} />
        <ContentWrapper boxClass="box-success">
          <div className="box-body">
            <FiltersWrapper />
            <hr />
            <div className="box-body">
              <div className="row">
                <div className="col-sm-12">
                  <AlertMessage />
                  <ErrorMessage />
                  <ReactTable
                    manual
                    className="-striped"
                    columns={[...this.columns]}
                    data={logs.list}
                    pages={logs.lastPage}
                    loading={loading}
                    onFetchData={this.onFetchData}
                    defaultPageSize={15}
                    PaginationComponent={Pagination}
                    multiSort={false}
                    resizable={false}
                    minRows={1}
                  />
                </div>
              </div>
            </div>
          </div>
        </ContentWrapper>
      </div>
    );
  }
}

const mapStateToProps = ({ logs, loading }) => ({ logs, loading });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(LogsList);
