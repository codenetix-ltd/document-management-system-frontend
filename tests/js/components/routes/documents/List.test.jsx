import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { DocumentsList as List } from 'Components/routes/documents/List';

describe('Documents list', () => {
  let spy;
  let props;
  let onSelectSpy;
  let onFetchDataSpy;
  let onArchiveSpy;
  let onDeleteSpy;
  let clearSelectionSpy;
  let wrapper;
  let instance;

  beforeEach(() => {
    spy = jest.fn();
    props = {
      documents: { list: [] },
      dispatch: spy,
      loading: false
    };
    onSelectSpy = jest.spyOn(List.prototype, 'onSelect');
    onFetchDataSpy = jest.spyOn(List.prototype, 'onFetchData');
    onArchiveSpy = jest.spyOn(List.prototype, 'onArchive');
    onDeleteSpy = jest.spyOn(List.prototype, 'onDelete');
    clearSelectionSpy = jest.spyOn(List.prototype, 'clearSelection');
    wrapper = shallow(<List {...props} />);
    instance = wrapper.instance();
    instance.prompt = {
      show: () => {}
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should call componentWillUnmount', () => {
    const { componentWillUnmount } = List.prototype;
    class ExtendedList extends List {
      componentWillUnmount = componentWillUnmount;
      render() {
        return <List {...props} />;
      }
    }
    const list = shallow(<ExtendedList {...props} />);
    list.unmount();
    expect(clearSelectionSpy.mock.calls.length).toBe(1);
  });

  it('should call clearSelection', () => {
    instance.clearSelection();
    expect(clearSelectionSpy.mock.calls.length).toBe(1);
  });

  it('should call onFetchData', () => {
    instance.onFetchData({
      page: 1,
      sorted: [{ id: 1, desc: true }]
    });
    expect(onFetchDataSpy.mock.calls.length).toBe(1);
  });

  it('should call onArchive', () => {
    instance.onArchive({
      value: {
        actualVersion: {}
      }
    });
    expect(onArchiveSpy.mock.calls.length).toBe(1);
  });

  it('should call onDelete', () => {
    instance.onDelete({
      value: {
        actualVersion: {}
      }
    });
    expect(onDeleteSpy.mock.calls.length).toBe(1);
  });

  it('should call onSelect', () => {
    instance.onSelect();
    expect(onSelectSpy.mock.calls.length).toBe(1);
  });
});
