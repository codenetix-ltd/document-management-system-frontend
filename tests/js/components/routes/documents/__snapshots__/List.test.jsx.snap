// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents list should render correctly 1`] = `
<div>
  <ContentHeader
    breadcrumbs={
      Array [
        Object {
          "iconCls": "fa fa-list",
          "pageLink": "/documents/list",
          "pageName": "Documents list",
        },
      ]
    }
    title="Documents"
  />
  <ContentWrapper
    boxClass="box-success"
    noBox={false}
  >
    <div
      className="box-body"
    >
      <Connect(FiltersWrapper) />
      <hr />
      <Connect(DocumentActions) />
      <div
        className="box-body"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-12"
          >
            <withRouter(Connect(AlertMessage)) />
            <withRouter(Connect(ErrorMessage)) />
            <SelectableTable
              PaginationComponent={[Function]}
              className="-striped"
              columns={
                Array [
                  Object {
                    "Header": "Id",
                    "accessor": "id",
                    "maxWidth": 65,
                  },
                  Object {
                    "Header": "Name",
                    "accessor": "actualVersion.name",
                  },
                  Object {
                    "Header": "Owner",
                    "accessor": "owner.fullName",
                  },
                  Object {
                    "Header": "Template ",
                    "accessor": "actualVersion.template.name",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Created at",
                    "accessor": "createdAt",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Updated at",
                    "accessor": "updatedAt",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "",
                    "maxWidth": 122,
                    "sortable": false,
                  },
                ]
              }
              data={Array []}
              defaultPageSize={15}
              loading={false}
              manual={true}
              multiSort={false}
              onFetchData={[Function]}
              onSelect={[Function]}
              resizable={false}
            />
          </div>
        </div>
      </div>
    </div>
  </ContentWrapper>
  <Prompt
    body=""
    cancelText="Cancel"
    confirmText="Confirm"
    headerText="Confirmation"
  />
</div>
`;
